bundle/Controller/ContentTypeController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RepositoryFormsBundle\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use eZ\Publish\Core\Repository\Values\ContentType\ContentTypeCreateStruct;
+use EzSystems\RepositoryForms\Data\Mapper\ContentTypeDraftMapper;
+use EzSystems\RepositoryForms\Event\FormActionEvent;
+use EzSystems\RepositoryForms\Event\RepositoryFormEvents;
+use Symfony\Component\HttpFoundation\Request;
+
+class ContentTypeController extends Controller
+{
+    public function createContentTypeAction($contentTypeGroupId, $languageCode = null)
+    {
+        $languageCode = $languageCode ?: $this->getConfigResolver()->getParameter('languages')[0];
+        $contentTypeService = $this->getRepository()->getContentTypeService();
+        $contentTypeGroup = $contentTypeService->loadContentTypeGroup($contentTypeGroupId);
+
+        $contentTypeCreateStruct = new ContentTypeCreateStruct([
+            'identifier' => 'new_content_type',
+            'mainLanguageCode' => $languageCode,
+            'names' => [$languageCode => 'New ContentType'],
+        ]);
+        $contentTypeDraft = $contentTypeService->createContentType($contentTypeCreateStruct, [$contentTypeGroup]);
+
+        return $this->redirectToRoute(
+            'contenttype/update',
+            ['contentTypeId' => $contentTypeDraft->id, 'languageCode' => $languageCode]
+        );
+    }
+
+    public function updateContentTypeAction(Request $request, $contentTypeId, $languageCode = null)
+    {
+        $languageCode = $languageCode ?: $this->getConfigResolver()->getParameter('languages')[0];
```

Comment:
Well, we'd have a lot of dependencies for very low gain IMHO.
Services needed:
* Templating
* Router
* FormFactory
* EventDispatcher
* ContentTypeService
* FieldTypeFormMapperRegistry

+ of course language list as injected setting.